README.md

```diff
@@ -0,0 +1,950 @@
+
+``` ipldsch
+-- Statements
+
+type Statement union {
+  | Equality
+  | Inequality
+  | Connective
+  | Negation
+  | Quantifier
+}
+
+-- Equality
+
+type EqOp enum {
+  | Eq ("==")
+  | Neq ("!=")
+}
+
+type Equality struct {
+  op EqOp
+  sel Selector
+  val Any
+} representation tuple
+
+type LikeOp enum {
+  | Like ("like")
+}
+
+type Like struct {
+  op LikeOp
+  sel Selector
+  str Wildcard
+} representation tuple
+
+-- Inequality
+
+type IneqOp enum {
+  | GT  (">")
+  | GTE (">=")
+  | LT  ("<")
+  | LTE ("<=")
+}
+
+type Inequality struct {
+  op IneqOp
+  sel Selector
+  val Number
+} representation tuple
+
+-- Connectives
+
+type NegateOp {
+  | Not ("not")
+}
+
+type Negation struct {
+  op   NegateOp
+  smts [Statement]
+} representation tuple
+
+type ConnectiveOp enum {
+  | And ("and")
+  | Or  ("or")
+}
+
+type Connective struct {
+  op   ConnectiveOp
+  smts [Statement]
+} representation tuple
+
+-- Quantification
+
+type QuantifierOp enum {
+  | All ("all")
+  | Any ("any")
+}
+
+type Quantifier struct {
+  op  QuantiefierOp
+  sel Selector
+  smt Statement
+} representation tuple
+
+-- Primitives
+
+type Selector = string
+
+type Number union {
+  | NumInt   int
+  | NumFloat float
+} representation kinded
+
+type Wildcard = string
+```
+ 
+## Comparisons
+[Comparisons]: #comparisons
+
+| Operator | Argument(s)                    | Example                          |
+|----------|--------------------------------|----------------------------------|
+| `==`     | `Selector, IPLD`               | `["==", ".a", [1, 2, {"b": 3}]]` |
+| `<`      | `Selector, (integer \| float)` | `["<",  ".a", 1]`                |
+| `<=`     | `Selector, (integer \| float)` | `["<=", ".a", 1]`                |
+| `>`      | `Selector, (integer \| float)` | `[">",  ".a", 1]`                |
+| `>=`     | `Selector, (integer \| float)` | `[">=", ".a", 1]`                |
+
+Literal equality (`==`) MUST match the resolved selecor to entire IPLD argument. This is a "deep comparison".
+
+Numeric inequalities MUST be agnostic to numeric type. In other words, the decimal representation is considered equivalent to an integer (`1 == 1.0 == 1.00`). Attempting to compare a non-numeric type MUST return false and MUST NOT throw an exception.
+
+## Glob Matching
+[Glob Matching]: #glob-matching
+
+| Operator | Argument(s)         | Example                               |
+|----------|---------------------|---------------------------------------|
+| `like`   | `Selector, Pattern` | `["like", ".email", "*@example.com"]` |
+
+Glob patterns MUST only include one specicial character: `*` ("wildcard"). There is no single character matcher. As many `*`s as desired MAY be used. Non-wildcard `*`-literals MUST be escaped (`"\*"`). Attempting to match on a non-string MUST return false and MUST NOT throw an exception.
```

Comment:
is zero `*` allowed? Does that means exactly one `*` or at most one `*`?

Reply:
+1

Reply:
For the next person, I read that wrong:

> Glob patterns MUST only include one special character

means: the grammar only has one special character (`*`), but that character is allowed multiple times in the same pattern: `foo*bar*baz` is valid.